packages/core/strapi/lib/core-api/controller/index.js

```diff
@@ -10,10 +10,15 @@ const createCollectionTypeController = require('./collection-type');
 
 const getAuthFromKoaContext = getOr({}, 'state.auth');
 
-const createController = ({ contentType }) => {
+const createController = ({ contentType, isCustom }) => {
   const ctx = { contentType };
 
   const proto = {
+
+    isCustom() {
```

Comment:
@Convly This was my idea to have it be the responsibility of the controller to know if it was customized or not, but I'm still not 100% convinced it's the best way.

Also, it's a very slight breaking change because users will no longer be able to have their own controller method called "isCustom" but I think that's acceptable. We could even make it something less likely to conflict like isControllerCustomized which I doubt would be breaking for anyone.

Reply:
Proposal: 

If the factory is used with customization set a non enumerable symbol we create on it (we do that for routing & auth stuff)  so it cannot conflict with user code.

Reply:
I like the idea